### ISSUE TITLE
iOS app 2024.9.2 An error has occurred

### ISSUE THREAD

**Issue by nathcoad** [dani-garcia/vaultwarden#5013]
### Your environment (Generated via diagnostics page)
* Vaultwarden version: v1.32.0
* Web-vault version: v2024.6.2b
* OS/Arch: linux/x86_64
* Running within a container: true (Base: Debian)
* Environment settings overridden: true
* Uses a reverse proxy: true
* IP Header check: false (X-Forwarded-For)
* Internet access: true
* Internet access via a proxy: false
* DNS Check: true
* Browser/Server Time Check: true
* Server/NTP Time Check: true
* Domain Configuration Check: true
* HTTPS Check: true
* Database type: SQLite
* Database version: 3.46.0
* Clients used: 
* Reverse proxy and version: 
* Other relevant information: 

### Config (Generated via diagnostics page)
<details><summary>Show Running Config</summary>

**Environment settings which are overridden:** DOMAIN, SIGNUPS_ALLOWED, SIGNUPS_VERIFY, INVITATIONS_ALLOWED, SHOW_PASSWORD_HINT, ADMIN_TOKEN, SMTP_HOST, SMTP_SECURITY, SMTP_PORT, SMTP_FROM, SMTP_USERNAME, SMTP_PASSWORD, SMTP_AUTH_MECHANISM, SMTP_ACCEPT_INVALID_CERTS

```json
{
  "_duo_akey": null,
  "_enable_duo": false,
  "_enable_email_2fa": true,
  "_enable_smtp": true,
  "_enable_yubico": true,
  "_icon_service_csp": "",
  "_icon_service_url": "",
  "_ip_header_enabled": true,
  "_max_note_size": 10000,
  "_smtp_img_src": "cid:",
  "admin_ratelimit_max_burst": 3,
  "admin_ratelimit_seconds": 300,
  "admin_session_lifetime": 20,
  "admin_token": "***",
  "allowed_iframe_ancestors": "",
  "attachments_folder": "data/attachments",
  "auth_request_purge_schedule": "30 * * * * *",
  "authenticator_disable_time_drift": false,
  "data_folder": "data",
  "database_conn_init": "",
  "database_max_conns": 10,
  "database_timeout": 30,
  "database_url": "***************",
  "db_connection_retries": 15,
  "disable_2fa_remember": false,
  "disable_admin_token": false,
  "disable_icon_download": false,
  "domain": "*****://***************",
  "domain_origin": "*****://***************",
  "domain_path": "",
  "domain_set": true,
  "duo_context_purge_schedule": "30 * * * * *",
  "duo_host": null,
  "duo_ikey": null,
  "duo_skey": null,
  "duo_use_iframe": false,
  "email_2fa_auto_fallback": false,
  "email_2fa_enforce_on_verified_invite": false,
  "email_attempts_limit": 3,
  "email_change_allowed": true,
  "email_expiration_time": 600,
  "email_token_size": 6,
  "emergency_access_allowed": true,
  "emergency_notification_reminder_schedule": "0 3 * * * *",
  "emergency_request_timeout_schedule": "0 7 * * * *",
  "enable_db_wal": true,
  "enable_websocket": true,
  "event_cleanup_schedule": "0 10 0 * * *",
  "events_days_retain": null,
  "experimental_client_feature_flags": "fido2-vault-credentials",
  "extended_logging": true,
  "helo_name": null,
  "hibp_api_key": null,
  "http_request_block_non_global_ips": true,
  "http_request_block_regex": null,
  "icon_blacklist_non_global_ips": true,
  "icon_blacklist_regex": null,
  "icon_cache_folder": "data/icon_cache",
  "icon_cache_negttl": 259200,
  "icon_cache_ttl": 2592000,
  "icon_download_timeout": 10,
  "icon_redirect_code": 302,
  "icon_service": "internal",
  "incomplete_2fa_schedule": "30 * * * * *",
  "incomplete_2fa_time_limit": 3,
  "increase_note_size_limit": false,
  "invitation_expiration_hours": 120,
  "invitation_org_name": "CoadCorp",
  "invitations_allowed": true,
  "ip_header": "X-Real-IP",
  "job_poll_interval_ms": 30000,
  "log_file": "/data/vaultwarden.log",
  "log_level": "debug",
  "log_timestamp_format": "%Y-%m-%d %H:%M:%S.%3f",
  "login_ratelimit_max_burst": 10,
  "login_ratelimit_seconds": 60,
  "org_attachment_limit": null,
  "org_creation_users": "",
  "org_events_enabled": false,
  "org_groups_enabled": false,
  "password_hints_allowed": true,
  "password_iterations": 100000,
  "push_enabled": false,
  "push_identity_uri": "https://identity.bitwarden.com",
  "push_installation_id": "***",
  "push_installation_key": "***",
  "push_relay_uri": "https://push.bitwarden.com",
  "reload_templates": false,
  "require_device_email": false,
  "rsa_key_filename": "data/rsa_key",
  "send_purge_schedule": "0 5 * * * *",
  "sendmail_command": null,
  "sends_allowed": true,
  "sends_folder": "data/sends",
  "show_password_hint": false,
  "signups_allowed": false,
  "signups_domains_whitelist": "",
  "signups_verify": true,
  "signups_verify_resend_limit": 6,
  "signups_verify_resend_time": 3600,
  "smtp_accept_invalid_certs": false,
  "smtp_accept_invalid_hostnames": false,
  "smtp_auth_mechanism": "Login",
  "smtp_debug": false,
  "smtp_embed_images": true,
  "smtp_explicit_tls": null,
  "smtp_from": "********************",
  "smtp_from_name": "Vaultwarden",
  "smtp_host": "*****************",
  "smtp_password": "***",
  "smtp_port": 587,
  "smtp_security": "starttls",
  "smtp_ssl": null,
  "smtp_timeout": 15,
  "smtp_username": "*********************",
  "templates_folder": "data/templates",
  "tmp_folder": "data/tmp",
  "trash_auto_delete_days": null,
  "trash_purge_schedule": "0 5 0 * * *",
  "use_sendmail": false,
  "use_syslog": false,
  "user_attachment_limit": null,
  "user_send_limit": null,
  "web_vault_enabled": true,
  "web_vault_folder": "web-vault/",
  "yubico_client_id": null,
  "yubico_secret_key": null,
  "yubico_server": null
}
```
</details>

### Vaultwarden Build Version

v1.32.0

### Deployment method

Official Container Image

### Reverse Proxy

caddy

### Host/Server Operating System

Linux

### Operating System Version

Ubuntu 22.04

### Clients

iOS

### Client Version

2024.9.2

### Steps To Reproduce

1. Open app
2. Logon or authenticate with face id. With 2FA enabled or not, same symptoms
3. After logon, see popup "an error has occurred" before showing any stored passwords

### Expected Result

My vault page should show stored data

### Actual Result

App displays popup "an error has occurred"

### Logs

```text
vaultwarden  | [2024-10-01 07:37:08.829][request][INFO] GET /api/sync
vaultwarden  | [2024-10-01 07:37:08.829][request][INFO] GET /api/sync
vaultwarden  | [2024-10-01 07:37:08.857][response][INFO] (sync) GET /api/sync?<data..> => 200 OK
vaultwarden  | [2024-10-01 07:37:08.858][response][INFO] (sync) GET /api/sync?<data..> => 200 OK
```

### Additional Context

Tried logging out on app and logging back in again.
Tried removing app completely and re-installing
Tried removing app completely, powering off iPhone, powering back on and re-installing app
Same symptoms shown every time.

**Comment by sebhofmann** [comment]
I have the same Issue.

**Comment by beyondwatts** [comment]
Also here, thanks for raising

**Comment by nathcoad** [comment]
Yep the older client on my iPad (still running iOS 17) works fine, as does the web client. It’s just this updated client in iOS 18 that has the issue.

I’m sure this means that something has changed in the latest version of app, but it is annoying the debug logs for the container don’t show any errors.

**Comment by BlackDex** [comment]
Please try the `testing` tagged images and see if that resolves the issue.

**Comment by BlackDex** [comment]
Also, Vaultwarden can't show debug logs of the clients of course. And the clients have an issue with the data received, and they are not very clear on what or why.

**Comment by nathcoad** [comment]
> Also, Vaultwarden can't show debug logs of the clients of course. And the clients have an issue with the data received, and they are not very clear on what or why.

Yep, totally understandable. I didn't mean to come across as blaming Vaultwarden for that, just that it makes it difficult to troubleshoot when I can't convey a more meaningful error message.


> Please try the `testing` tagged images and see if that resolves the issue.

Just tried updating my docker-compose.yaml and running a `docker compose down && docker compose up --force-recreate -d` and now I can login using the new iOS client. Server is now running 1.32.0-040e2a7b

Thankyou so much!

**Comment by Vip0r** [comment]
Yes - can confirm that this issue appeared on my side as well since i updated my iphone ios client app

**Comment by jmehitch** [comment]
I had the same issue, updating to image version 1.32.0 worked for me too.

**Comment by BlackDex** [comment]
Great! This does mean we are closer to a new release, since a lot of these issues should be solved now.
Thanks for testing.

Going to close this then as working with the latest testing images.

**Comment by beyondwatts** [comment]
1.32.0 worked for me. thank you for the speedy fix!

**Comment by Al3x360** [comment]
Hi, just wanted to share you an info : i tried Testing, and it worked : However, I use authentik to protect our self-hosted server . the auth cant pass to authentik 

[2024-10-03 16:42:07.473][vaultwarden::api::identity][ERROR] Username or password is incorrect. Try again. IP: docker_ip_addr. Username: user@exemple.com.
[2024-10-03 16:42:07.473][response][INFO] (login) POST /identity/connect/token => 400 Bad Request

If i disable Authentik in the nginix reverse proxy the authentication works from iOS. 

I've setup the API call bypass in Authentik using regex : 

this is set up here in Authentik : Proxy provider >Advanced protocol parameters > non authenticated paths

^/api([/?].*)?$
^/identity.*$
^/triggers.*$
^/meshagents.*$
^/meshsettings.*$
^/agent.*$
^/control.*$
^/meshrelay.*$
^/wl.*$

You need to do this for bitwarden app to communicate with vaultwarden behind authentik authentication service. Maybe there are new regex to update regarding the api section ?

**Comment by stonebyte** [comment]
I had the "An error has occured" problem in the Bitwarden iOS app as well. 
Vaultwarden docker image runs on a small server behind my Internet router. Docker image was usually `latest`, I switched to `testing` while hunting down the app problem. iOS was updated to latest available as well.

While hunting for the root cause, I tried accessing Vaultwarden web UI from my iPhone and noticed the following behaviour:
- DuckDuckGo Browser -> connection failed
- Brave Browser -> connection failed
- Apple Safari Browser -> works

This means, there is not only a problem with Bitwarden App but also third party browsers have an issue connecting to my vaultwarden instance from iPhone.

Next I disabled WiFi and used cellular bearer for accessing web UI:
- all browsers work

As I run a split horizon DNS setup for my selfhosted services, this is a strong indicator for a DNS problem.

But why the different behaviour between Safari and the third party browsers? If an Apple SW behaves diffently from third party SW, another Apple service could be the culprit. I remembered that I enabled "Private Relay" some weeks ago, which ~~messes around with DNS~~ uses ODoH and thus buypasses my local DNS. 

What seems to happen is, that third pary apps only use Apples DNS Servers when private relay is eabled and hence only get the public IP address which isn´t reachable from the inside. Safari seems to behave differently, no idea why.

TL;DR: 
**It's always DNS. Disable Apple Private Relay in case you have it enabled and run into "An error has occured" problems with bitwarden iOS app.**

If you really want to use Apple private relay with split horizon DNS, you will have to configure pinhole NAT on your router.
